### PR TITLE
Various pmi fixes

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Get things working again with netcoreapp5_0.

* updated and generalize some of the conditional compilation symbols
* log driveall invocations
* update help text
* add handler for a common exception type

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
Should fix #237. Will update to also take care of #238.

cc @dotnet/jit-contrib

File: src/pmi/pmi.cs [hunk 1]
```diff
@@ -122,7 +122,7 @@ public static Assembly Resolve(string assemblyName, AssemblyLoadContext context)
     }
 }
 
-#if NETCOREAPP3_0 || NETCOREAPP3_1
+#if NETCOREAPP3_0 || NETCOREAPP3_1 || NETCOREAPP5_0
```
Comment (BruceForstall) [inline src/pmi/pmi.cs]: Couldn't we reverse this, so the netcoreapp2.2 (?) and below (in the `else` clause) get the `#if` and 3.0+ get the `#else` and thus don't need to get updated going forward?

File: src/pmi/pmi.cs [hunk 2]
```diff
@@ -1270,28 +1275,30 @@ private static int Usage()
...
-            + "      The same as PrepAll, but is more robust. While PrepAll will stop at the first JIT assert, DriveAll will\r\n"
-            + "      continue by skipping that method.\r\n"
-            + "\r\n"
-            + "Environment variable PMIPATH is a semicolon-separated list of paths used to find dependent assemblies.\r\n"
-            + "\r\n"
-            + "Optional suffixes will change behavior:\r\n"
-            + "   -CCtors will jit and run cctors before jitting other methods\r\n"
-            + "   -Quiet will suppress in-progress messages for type and method exploration\r\n"
-            + "   -Time will always show elapsed times, even in -Quiet mode"
+            "Usage:\n"
```
Comment (BruceForstall) [inline src/pmi/pmi.cs]: I presume Console.WriteLine does embedded `\n` => `\r\n` translation on Windows? (but not on Unix)

File: src/pmi/pmi.cs (hunk 1 above)
Comment (AndyAyersMS) [reply src/pmi/pmi.cs]: Sure.

File: src/pmi/pmi.cs (hunk 2 above)
Comment (AndyAyersMS) [reply src/pmi/pmi.cs]: I can't find anything official, but it works fine; I've never actually used explicit `\r\n`.

Comment (BruceForstall) [reply src/pmi/pmi.cs]: I presume explicit `\r\n` isn't great on Unix.

**Pull request merged**